dissect/cstruct/types/bytesinteger.py

```diff
@@ -38,9 +38,22 @@ def parse(buf: BinaryIO, size: int, count: int, signed: bool, endian: str) -> Li
         return nums
 
     @staticmethod
-    def pack(data: List[int], size: int, endian: str) -> bytes:
+    def pack(data: List[int], size: int, endian: str, signed: bool) -> bytes:
         buf = []
+
+        bits = size * 8
+        unsigned_min = 0
+        unsigned_max = (2**bits) - 1
+        signed_min = -(2 ** (bits - 1))
+        signed_max = (2 ** (bits - 1)) - 1
+
         for i in data:
+
+            if signed and (i < signed_min or i > signed_max):
+                raise OverflowError()
```

Comment:
```suggestion
                raise OverflowError(f"{i} exceeds bounds for signed {bits} bits BytesInteger")
```
Maybe add an error message long these lines? Same for the one below.

Reply:
OK